### ISSUE TITLE
Implement learning

### ISSUE THREAD

**Issue by gustav1101** [vab9/pir-project2017#22]
- [x] Prepare Backpropagation which will use Gradient Descent
- [x] Implement Gradient Descent
